splice/static/js/controllers/upcoming.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+angular.module('spliceApp').controller('upcomingController', function($controller, $scope, $modal, spliceData, initData) {
+  $controller('distributionController', {$scope: $scope});
+
+  $scope.unscheduleErrorMsg = null;
+  $scope.previewModal = $modal({scope: $scope, template: "template/preview.html", show: false});
+
+  /** UI and tile data **/
+
+  $scope.previewDist = function(url) {
+    $scope.openRemoteDistribution(url);
+    $scope.previewModal.$promise.then($scope.previewModal.show);
+  };
+
+  $scope.unscheduleDist = function(id) {
+    if (window.confirm("Really unschedule distribution " + id + "?")) {
+      spliceData.unscheduleDistribution(id)
+        .success(function(data) {
+          $scope.refreshDistributions();
+        })
+        .error(function(data, status, headers, config, statusText) {
+            $scope.unscheduleErrorMsg = '<p>Could not unschedule distribution ' + id + '</p>';
+            $scope.unscheduleErrorMsg += '<p>HTTP Error: ' + status + ' ' + statusText + '</p>';
+            $scope.unscheduleErrorMsg += '<p>HTTP Error: ' + status + ' ' + statusText + '</p>';
+            if (data && data.message) {
+              $scope.unscheduleErrorMsg += '<p>' + data.message + '</p>';
+            }
+        });
+    }
+  };
+
+  $scope.openRemoteDistribution = function(url) {
+    /**
+     * Open a distribution file given a URL
+     */
+    var chanId = $scope.channelSelect;
+    var cacheKey = chanId + url;
+
+    if (!$scope.cache.hasOwnProperty(cacheKey)) {
```

Comment:
This entire if/else statement seems awfully similar to the one in authoring.js in `$scope.$watch('versionSelect, ...)` Both are fetching and loading distribution json. Perhaps we can factor out some of this into a shared function?